x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -130,13 +143,18 @@ function transformResponseToServiceAnomalies(
     response.aggregations?.services.buckets ?? []
   ).reduce(
     (statsByServiceName, { key: serviceName, top_score: topScoreAgg }) => {
+      const mlResult = maybe(topScoreAgg.hits.hits[0])?._source;
+
+      const jobId = mlResult?.job_id;
+      const transactionType = mlResult?.by_field_value;
+
       return {
         ...statsByServiceName,
         [serviceName]: {
-          transactionType: topScoreAgg.hits.hits[0]?._source?.by_field_value,
-          anomalyScore: topScoreAgg.hits.hits[0]?.sort?.[0],
-          actualValue: topScoreAgg.hits.hits[0]?._source?.actual?.[0],
-          jobId: topScoreAgg.hits.hits[0]?._source?.job_id,
+          transactionType,
+          jobId,
+          actualValue: mlResult?.actual?.[0],
+          anomalyScore: mlResult?.record_score || 0,
```

Comment:
nit: any reason some of these are inlined and others are not?
```suggestion
          transactionType: mlResult?.job_id,
          jobId: mlResult?.by_field_value,
          actualValue: mlResult?.actual?.[0],
          anomalyScore: mlResult?.record_score || 0,
```